src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -325,14 +325,35 @@ std::string ContextAttribute::renderAsJsonObject
   {
     if (omitValue == false)
     {
+      std::string effectiveValue = "";
+      bool valueIsNumberOrBool = false;
+      switch(valueType)
+      {
+      case ValueTypeString:
+        effectiveValue = stringValue;
+        break;
+      case ValueTypeBoolean:
+        effectiveValue = boolValue? "true" : "false";
+        valueIsNumberOrBool = true;
+        break;
+      case ValueTypeNumber:
+        char num[32];
```

Comment:
Does this work?
Ah, of course, its C++ ... you can declare variables wherever you want ... :-(